mesonbuild/build.py

```diff
@@ -974,9 +974,15 @@ def _extract_pic_pie(self, kwargs, arg):
             mlog.warning("Use the '{}' kwarg instead of passing '{}' manually to {!r}".format(arg, '-f' + arg, self.name))
             return True
 
-        val = kwargs.get(arg, False)
-        if not isinstance(val, bool):
-            raise InvalidArguments('Argument {} to {!r} must be boolean'.format(arg, self.name))
+        val = kwargs.get(arg, None)
+        if val == "default":
+            val = None
+        elif val == 'true':
+            val = True
+        elif val == 'false':
+            val = False
+        if val not in {True, False, None}:
+            raise InvalidArguments('Argument {} to {!r} must one of "true", "false", "default"'.format(arg, self.name))
```

Comment:
I don't think this part is correct.
- That's used to extract `pic` and `pie` from i.e. `executable()` keyword arguments. Possible values are bool or None, it must not be string 'true'.
- Do we want to support tristate for PIC too? So far this PR is only about PIE.
- You doc and changelog only mention change to the `b_pie` builtin option, not those keyword arguments. But I guess you're right that `executable(..., pie: false)` should produce `-no-pie` arg.

Reply:
I don't know about PIC, I've updated the documents to mention the change to the meson language

Reply:
I've also fixed the pie vs pic differences. Maybe it does make sense to do the same for PIC?

Reply:
We can handle the PIC case in another PR if anyone complains.